### ISSUE TITLE
ampliconclip multiple chrom bug

### ISSUE THREAD

**Issue by nathanhaigh** [samtools/samtools#1418]
@whitwham 

Running samtools v1.12 on:

- Linux 4.9.0-5-amd64
- x86_64
- under conda

`test.sam` with read alignments on `chr1`:
```
@HD     VN:1.6  SO:coordinate
@SQ     SN:chr1 LN:10
@SQ     SN:chr2 LN:30
read1   99      chr1    1       60      4M      chr1    7       10      AAAA    HHHH
read1   147     chr1    7       60      4M      chr1    1       10      TTTT    HHHH
```

`test5.bed` with regions defined on `chr2`:
```
chr2    0       1       region1 0       +
chr2    9       10      region2 0       -
```

testing:
```
samtools ampliconclip --original -O SAM -b test5.bed test.sam
COMMAND: samtools ampliconclip -b test5.bed test.sam
TOTAL READS: 2
TOTAL CLIPPED: 2
FORWARD CLIPPED: 1
REVERSE CLIPPED: 1
BOTH CLIPPED: 0
NOT CLIPPED: 0
EXCLUDED: 0
FILTERED: 0
FAILED: 0
WRITTEN: 2
@HD     VN:1.6  SO:unknown
@SQ     SN:chr1 LN:10
@SQ     SN:chr2 LN:30
@PG     ID:samtools     PN:samtools     VN:1.12 CL:samtools ampliconclip --original -b test5.bed test.sam
read1   99      chr1    2       60      1S3M    =       7       10      AAAA    HHHH    OA:Z:read1,1,+,4M60,;
read1   147     chr1    7       60      3M1S    =       1       10      TTTT    HHHH    OA:Z:read1,7,-,4M60,;
```

Expected output:
NOT to clip `read1` alignments as they are on `chr1` NOT `chr2`

**Comment by whitwham** [comment]
Code was merged in to do this (see #1410) a few days ago.  Does it fail with the new code?

**Comment by whitwham** [comment]
Closing after no response for a week.
